### PR TITLE
fix(list-key-manager): infinite loop if all items are disabled

### PR DESCRIPTION
**Pull request by DevVersion**
* Currently if the `ListKeyManager` is being used in wrap mode, and all items are disabled, pressing the down arrow key will cause the key manager to go into an infinite loop that searches for the next item.

Related #9963

### CODE REVIEW

File: src/cdk/a11y/key-manager/list-key-manager.spec.ts [hunk 1]
```diff
@@ -467,6 +467,16 @@ describe('Key managers', () => {
         expect(keyManager.setActiveItem).toHaveBeenCalledWith(0);
       });
 
+      // This test should pass if all items are disabled and the down arrow key got pressed.
+      // If the test setup crashes or this test times out, this test can be considered as failed.
+      it('should not get into an infinite loop if all items are disabled', () => {
+        keyManager.withWrap();
+        keyManager.setActiveItem(0);
```
Comment (crisbeto) [inline src/cdk/a11y/key-manager/list-key-manager.spec.ts]: You can chain the `withWrap` and `setActiveItem`.

File: src/cdk/a11y/key-manager/list-key-manager.ts [hunk 2]
```diff
@@ -269,16 +269,15 @@ export class ListKeyManager<T extends ListKeyManagerOption> {
...
-    } else {
-      this.setActiveItem(this._activeItemIndex);
+  private _setActiveInWrapMode(delta: -1 | 1, items: T[]): void {
+    for (let i = 1; i <= items.length; i++) {
+      const index = (this._activeItemIndex + delta * i + items.length) % items.length;
+      const item = items[index];
+
+      if (!item.disabled) {
+        this.setActiveItem(index);
+        break;
```
Comment (crisbeto) [inline src/cdk/a11y/key-manager/list-key-manager.ts]: Might be a good idea to `return` here instead so we don't put some extra logic below by accident.

Comment (DevVersion) [reply src/cdk/a11y/key-manager/list-key-manager.ts]: Yeah that works too. Changed it

File: src/cdk/a11y/key-manager/list-key-manager.spec.ts (hunk 1 above)
Comment (DevVersion) [reply src/cdk/a11y/key-manager/list-key-manager.spec.ts]: I tried to keep it consistent with the rest of the tests. All of those don't seem to chain either.

File: src/cdk/a11y/key-manager/list-key-manager.ts [hunk 3]
```diff
@@ -269,16 +269,15 @@ export class ListKeyManager<T extends ListKeyManagerOption> {
...
-      (this._activeItemIndex + delta + items.length) % items.length;
-
-    // skip all disabled menu items recursively until an enabled one is reached
-    if (items[this._activeItemIndex].disabled) {
-      this._setActiveInWrapMode(delta, items);
-    } else {
-      this.setActiveItem(this._activeItemIndex);
+  private _setActiveInWrapMode(delta: -1 | 1, items: T[]): void {
+    for (let i = 1; i <= items.length; i++) {
+      const index = (this._activeItemIndex + delta * i + items.length) % items.length;
```
Comment (jelbourn) [inline src/cdk/a11y/key-manager/list-key-manager.ts]: Use parentheses around the multiplication to make the order of operations explicit

Comment (DevVersion) [reply src/cdk/a11y/key-manager/list-key-manager.ts]: Done.

**Pull request merged**